source/controllers/include/controllers/exceptions/InvalidControllerException.hpp

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <stdexcept>
+#include <string>
+
+namespace controllers {
+namespace exceptions {
+class InvalidControllerException : public std::logic_error {
+public:
+  explicit InvalidControllerException(const std::string& msg) : std::logic_error(msg){};
+};
+}// namespace exceptions
+}// namespace controllers
```

Comment:
Aren't we doing it like that?
```suggestion
namespace controllers::exceptions {
class InvalidControllerException : public std::logic_error {
public:
  explicit InvalidControllerException(const std::string& msg) : std::logic_error(msg){};
};
}// namespace controllers::exceptions
```